wallet/src/Cardano/Wallet/WalletLayer/Kernel.hs

```diff
@@ -153,6 +173,27 @@ bracketPassiveWallet pm mode logFunction keystore node fInjects f = do
         invokeIO :: forall m'. MonadIO m' => Actions.WalletAction Blund -> m' ()
         invokeIO = liftIO . STM.atomically . invoke
 
+xqueryWalletBalance :: Kernel.PassiveWallet -> EncryptedSecretKey -> IO WalletBalance
+xqueryWalletBalance w esk = do
+  let
+    nm = makeNetworkMagic (w ^. walletProtocolMagic)
+    wId    = WalletIdHdRnd (eskToHdRootId nm esk)
+    wKey :: WalletKey
+    wKey = (wId, eskToWalletDecrCredentials nm esk)
+    withNode :: (HasCompileInfo, HasUpdateConfiguration) => Lock (WithNodeState IO) -> WithNodeState IO Utxo
+    withNode _lock = getAllPotentiallyHugeUtxo
```

Comment:
I'd recommend to use `filterUtxo` (from `Cardano.Wallet.Kernel.NodeStateAdaptor`)  with something like `isOurs` (from `Cardano.Wallet.Kernel.HdWallet`) which would also to browse the current UTxO in a streaming fashion and not destroy your machine's memory :ok_hand:

Reply:
ah, that sounds much better